### ISSUE TITLE
Azure managed-groups role assignment not working

### ISSUE THREAD

**Issue by RobertSkawinski** [hashicorp/boundary#2782]
Hi all,

I'm deplyoing boundary with terraform.
The creation of managed-groups and role assignment seem to work, but the User is not able to manage the scopes.

The User DEV1 Boundary has no permissions in Scope ID: o_pzn0PkxDyQ
Boundary version 0.11.2

```
$ boundary auth-methods list 

...

Auth Method information:
  ID:                     amoidc_sgneGNsCtD
    Version:              2
    Type:                 oidc
    Name:                 Azure
    Description:          OIDC auth method for Azure
    Is Primary For Scope: true
    Authorized Actions:
      no-op
      read
      update
      delete
      change-state
      authenticate

```

```
$ boundary users list

...

  ID:                    u_oZUKlMmdlh
    Version:             1
    Primary Account ID:  acctoidc_XINTxVRk8g
    Login Name:          O81zDau9yvZiJcReTZviQo26JLFlFJlQSco6zbaUBtk
    Full Name:           DEV1 Boundary
    Authorized Actions:
      no-op
      read
      update
      delete
      add-accounts
      set-accounts
      remove-accounts
```

```
$ boundary managed-groups list -auth-method-id amoidc_sgneGNsCtD 

Managed Group information:

...

  ID:                    mgoidc_iXyHRFPLNY
    Version:             1
    Type:                oidc
    Name:                Azure Admins
    Description:         Azure Admin team managed group
    Authorized Actions:
      no-op
      read
      update
      delete
```
```
$ boundary managed-groups read -id mgoidc_iXyHRFPLNY

Managed Group information:
  Auth Method ID:      amoidc_sgneGNsCtD
  Created Time:        Thu, 12 Jan 2023 16:41:59 CET
  Description:         Azure Admin team managed group
  ID:                  mgoidc_iXyHRFPLNY
  Name:                Azure Admins
  Type:                oidc
  Updated Time:        Thu, 12 Jan 2023 16:41:59 CET
  Version:             1

  Scope:
    ID:                global
    Name:              global
    Type:              global

  Authorized Actions:
    no-op
    read
    update
    delete

  Attributes:
    Filter:            "c46511ec-xxx-8def72856260" in "/token/groups"
```

```
$ boundary roles list 
...

Role information:
  ID:                    r_eB9lbGJMtw
    Version:             4
    Name:                Infra-admin
    Description:         Administrator role for Infra
    Authorized Actions:
      no-op
      read
      update
      delete
      add-principals
      set-principals
      remove-principals
      add-grants
      set-grants
      remove-grants

```

```
$ boundary roles read -id r_eB9lbGJMtw 

Role information:
  Created Time:        Thu, 12 Jan 2023 17:03:02 CET
  Description:         Administrator role for Infra
  Grant Scope ID:      o_pzn0PkxDyQ
  ID:                  r_eB9lbGJMtw
  Name:                Infra-admin
  Updated Time:        Thu, 12 Jan 2023 17:20:21 CET
  Version:             4

  Scope:
    ID:                global
    Name:              global
    Type:              global

  Authorized Actions:
    no-op
    read
    update
    delete
    add-principals
    set-principals
    remove-principals
    add-grants
    set-grants
    remove-grants

  Principals:
    ID:             mgoidc_iXyHRFPLNY
      Type:         managed group
      Scope ID:     global
    ID:             g_FGwvEOxZyu
      Type:         group
      Scope ID:     global

  Canonical Grants:
    id=*;type=*;actions=*

```
![image](https://user-images.githubusercontent.com/6197076/212131425-fa702799-6f98-4b80-b672-ae026fcd0f49.png)

**Comment by xingluw** [comment]
Hi @tritonblaster, each scope has its own set of distinct roles, so a user would need to be added to a role in each scope. For example:

Global has a global_admin role
Org has an org_admin role
Project has a project_admin role

If you had 10 different scopes and you want the same user to be able to view any of them, they would need to be added to 10 roles.

**Comment by tritonblaster** [comment]
Hi @xingluw,
Thank you for your response.
Yes I know. - in my example you can see the config and screenshot of scope o_pzn0PkxDyQ
In the screenshot you can see that all options on the left side are missing.

**Comment by tritonblaster** [comment]
The global scope is, where the auth method is located.
The role should be valid for the Infra scope

Grant Scope ID:      o_pzn0PkxDyQ

**Comment by xingluw** [comment]
Yes, the auth method, user, and managed group exists at the global level, and so does the role r_eB9lbGJMtw.

However in order to reach the Org scope for Infra, you will need a new role at the Infra level.

If you log in with your global super-admin user, and try going to Orgs -> Infra -> Roles, do you see any roles listed there? You will need one for that managed group or any users. If you did not use TF to set up Boundary and just used the standard setup, you will see that Boundary created roles at every level for you already.

**Comment by tritonblaster** [comment]
Hi @xingluw,

I tested your approach without success:

For local user with authentication password it is working with both configurations (role located in global scope and role located in affected scope)
For Azure AD user that is member of managed group it is neither working if the role is located in global scope nor the role is located in affected scope


Updated terraform config
```
resource "boundary_role" "org_admin" {
  name           = "${boundary_scope.org_infra.name}-admin"
  description    = "Administrator role for ${boundary_scope.org_infra.name}"
  scope_id       = boundary_scope.org_infra.id
  grant_scope_id = boundary_scope.org_infra.id
  grant_strings = [
    "id=*;type=*;actions=*"
  ]
  principal_ids = [
    boundary_group.admins.id,
    boundary_managed_group.azuread_admins.id
  ]
}
```
boundary config output:
```
$ boundary roles list -recursive

Role information:

...

  ID:                    r_XeuY7S6x7D
    Scope ID:            o_pzn0PkxDyQ
    Version:             3
    Name:                Infra-admin
    Description:         Administrator role for Infra
    Authorized Actions:
      no-op
      read
      update
      delete
      add-principals
      set-principals
      remove-principals
      add-grants
      set-grants
      remove-grants
```
```
$ boundary roles read -id r_XeuY7S6x7D

Role information:
  Created Time:        Fri, 13 Jan 2023 11:10:32 CET
  Description:         Administrator role for Infra
  Grant Scope ID:      o_pzn0PkxDyQ
  ID:                  r_XeuY7S6x7D
  Name:                Infra-admin
  Updated Time:        Fri, 13 Jan 2023 11:10:33 CET
  Version:             3

  Scope:
    ID:                o_pzn0PkxDyQ
    Name:              Infra
    Parent Scope ID:   global
    Type:              org

  Authorized Actions:
    no-op
    read
    update
    delete
    add-principals
    set-principals
    remove-principals
    add-grants
    set-grants
    remove-grants

  Principals:
    ID:             mgoidc_iXyHRFPLNY
      Type:         managed group
      Scope ID:     global
    ID:             g_FGwvEOxZyu
      Type:         group
      Scope ID:     global

  Canonical Grants:
    id=*;type=*;actions=*
```

**Comment by RobertSkawinski** [comment]
I discovered one more thing:

The web ui shows the user as member of the managed group, but the boundary cli does not show any member IDs.

![image](https://user-images.githubusercontent.com/6197076/212387991-c6cf0c63-f013-4070-9581-79360a6d378d.png)

```
$ boundary managed-groups read -id mgoidc_iXyHRFPLNY 

Managed Group information:
  Auth Method ID:      amoidc_sgneGNsCtD
  Created Time:        Thu, 12 Jan 2023 16:41:59 CET
  Description:         Azure Admin team managed group
  ID:                  mgoidc_iXyHRFPLNY
  Name:                Azure Admins
  Type:                oidc
  Updated Time:        Thu, 12 Jan 2023 16:41:59 CET
  Version:             1

  Scope:
    ID:                global
    Name:              global
    Type:              global

  Authorized Actions:
    no-op
    read
    update
    delete

  Attributes:
    Filter:            "c46511ec-xxx-260" in "/token/groups"
```

**Comment by xingluw** [comment]
Yes, right now the Admin UI does not show managed groups, that is being fixed for the next release.

Can I see how you defined your `boundary_managed_group.azuread_admins` in Terraform?

Also, can you verify that your managed group has your targeted account? This can be done using `boundary managed-groups read -d <managed-group-id>` and looking under "Member IDs:"

**Comment by xingluw** [comment]
Yes, right now the Admin UI does not show managed groups, that is being fixed for the next release.

I am seeing that your group is missing members (it should list them in your `managed-groups read` call). Can you make sure that your auth method configuration has "email", "groups", and "profile" under Claims Scopes? After that, check that an individual account returns group claims with `boundary accounts read -id <account-id>`

Here is an example:

```
Account information:
  Auth Method ID:      amoidc_xxx
  Created Time:        Mon, 16 Jan 2023 12:42:24 PST
  ID:                  acctoidc_xxx
  Type:                oidc
  Updated Time:        Mon, 16 Jan 2023 12:43:32 PST
  Version:             2

  Scope:
    ID:                global
    Name:              global
    Type:              global

  Authorized Actions:
    no-op
    read
    update
    delete

  Attributes:
    email:             xxx@gmail.com
    full_name:         My Name
    issuer:            https://xxx.okta.com
    subject:         xxx
    ...
    "groups": [
    "Everyone",
    "Test"
    ],
    ...
    }
```

After you have verified this, the filter for the managed group should be something like this:
`"Test" in "/userinfo/groups"`

![image](https://user-images.githubusercontent.com/95828202/212771057-7aab74a9-c75b-4f47-8d36-d874c75cc82e.png)

**Comment by RobertSkawinski** [comment]
Well, the interresting part ist, that I can see the correct user assignment in the admin ui but not in the Boundary CLI:
Selecting Auth Methods
![image](https://user-images.githubusercontent.com/6197076/212777815-cbf2ce84-4e59-438c-a15e-02bae6d9f9f3.png)
then the Azure OIDC Auth method
![image](https://user-images.githubusercontent.com/6197076/212777872-88e29ea8-b618-420c-be6d-d2525185a5f3.png)
and then switch to Managed Groups, I can see both managed groups
![image](https://user-images.githubusercontent.com/6197076/212777964-f9715188-35f3-4017-a2d4-eab89d6dbe90.png)
Then, when I enter the Admin Group, I can see the correct user assigned
![image](https://user-images.githubusercontent.com/6197076/212778033-37b208b8-03c9-45cc-b10b-411128ae36ae.png)

**Comment by xingluw** [comment]
It is strange that your UI shows managed group members, but your CLI does not. I am assuming that both are logged in with the same global admin account?

I will ask the team to investigate, what version of Boundary are you running, and is this an open source or HCP Boundary deployment?

**Comment by RobertSkawinski** [comment]
yes, both queries are done with the same admin user. (Admin UI & boundary cli)

I'm using boundary docker version 0.11.2
https://hub.docker.com/layers/hashicorp/boundary/0.11.2/images/sha256-ed6190e9722e81191f8d5a683e2ce3c69a0d2388e0c1f9b989a5bc26fd32ab4a?context=explore

Thanks for your support :)

If you need any detailed Information we can also schedule a Teams Session.

**Comment by Skesov** [comment]
Have some problem with gitlab oidc

**Comment by jefferai** [comment]
@Skesov Same problem meaning shows up in UI and not in CLI?

**Comment by Skesov** [comment]
I think the problem is bigger, than just UI. Managed groups is broken: don't work in roles and show weather...

```bash 
boundary accounts read -id acctoidc_UcWJd3bmwO

Account information:
  Auth Method ID:      amoidc_nQ3L3GlbX9
  Created Time:        Wed, 18 Jan 2023 18:58:11 MSK
  ID:                  acctoidc_UcWJd3bmwO
  Type:                oidc
  Updated Time:        Wed, 18 Jan 2023 19:37:42 MSK
  Version:             4

  Scope:
    ID:                global
    Name:              global
    Type:              global

  Authorized Actions:
    no-op
    read
    update
    delete

  Attributes:
    full_name:         Vladimir Skesov
    issuer:            https://git.omprussia.ru
    subject:           2222
    token_claims:
    {
    "aud": "222222222222222afb030f2a1e19cc603a52048b22222222222",
    "auth_time": 1674059823,
    "exp": 1674059981,
    "groups_direct": [
    "ocs-infra",
    "go-shared",
    "appstore",
    "authorization/obs-promo",
    "auth",
    "cdn",
    "cloud",
    "js-shared",
    "emm",
    "os",
    "pkgrepo",
    "infra/terra/examples",
    "sfc-sandbox",
    "push",
    "mt",
    "omp-qa",
    "ocs-qa",
    "mer-core",
    "nemomobile",
    "mer-hybris",
    "sailfishos",
    "customer-omp",
    "jolla",
    "omp_alldevops",
    "mer-packages",
    "mer-crosshelpers",
    "mer-tools",
    "jolla-p4903",
    "openmobileplatform",
    "show",
    "test_group_x",
    "omp-sdk",
    "qt",
    "omp-browser",
    "l10n",
    "libreoffice",
    "popul-census",
    "devops",
    "native-components",
    "libhybris",
    "mashtab_t1",
    "non-oss",
    "docs",
    "mer-qt",
    "mer-obs",
    "mer-core-attic",
    "mer-qa",
    "mer-core-contrib",
    "qt-creator",
    "qt-labs",
    "common-infra/public",
    "release_management",
    "authorization/open-build-service",
    "authorization/cve-search",
    "external_portal",
    "external_devdoc"
    ],
    "iat": 1674059861,
    "iss": "https://git.omprussia.ru",
    "nonce": "22222222222222222222",
    "sub": "2222",
    "sub_legacy": "2131313333333333333333333333333112312312312312365123123"
    }
    userinfo_claims:
    {
    "groups": [
    "docs",
    "cloud",
    "mer-core",
    "nemomobile",
    "mer-hybris",
    "sailfishos",
    "customer-omp",
    "jolla",
    "mer-qt",
    "mer-packages",
    "mer-crosshelpers",
    "mer-obs",
    "mer-core-attic",
    "mer-tools",
    "os",
    "jolla-p4903",
    "mer-qa",
    "docs/hardware",
    "docs/platform",
    "docs/sailfish_guide",
    "docs/instructions",
    "mer-core-contrib",
    "openmobileplatform",
    "docs/cloud",
    "qt-creator",
    "qt-labs",
    "omp-qa",
    "show",
    "go-shared",
    "emm",
    "sfc-sandbox",
    "omp-sdk",
    "appstore",
    "devops",
    "ocs-infra",
    "qt",
    "auth",
    "omp-browser",
    "js-shared",
    "pkgrepo",
    "l10n",
    "ocs-infra/mobile-ci",
    "cdn",
    "libreoffice",
    "push",
    "popul-census",
    "omp-sdk/components",
    "common-infra/public",
    "native-components",
    "ocs-infra/docker",
    "ocs-infra/pipeline",
    "libhybris",
    "omp-sdk/external",
    "os/kcapi",
    "release_management",
    "mt",
    "os/demo-apps",
    "mashtab_t1/droid",
    "external_devdoc",
    "external_devdoc/tools",
    "external_devdoc/omegat-projects",
    "authorization/open-build-service",
    "os/pns",
    "infra/terra/examples",
    "ocs-qa",
    "omp-qa/app_management_packages",
    "os/cryptocontainers",
    "os/appmanagement",
    "external_portal",
    "os/experimental",
    "omp-qa/pkg-fuzzing",
    "omp-qa/rpm-validator",
    "omp-qa/sdjd",
    "ocs-infra/template",
    "non-oss",
    "ocs-infra/mirror",
    "ocs-infra/mirror/docker-io",
    "ocs-infra/mirror/docker-io/library",
    "ocs-qa/alpetrov",
    "test_group_x",
    "authorization/cve-search",
    "qtech-qmp-m1-n/sunritel/vendor/libs",
    "omp-qa/benchmarks",
    "omp_alldevops",
    "external_portal/django",
    "authorization/obs-promo"
    ],
    "https://gitlab.org/claims/groups/developer": [
    "docs",
    "qt-creator",
    "qt-labs",
    "common-infra/public"
    ],
    "https://gitlab.org/claims/groups/maintainer": [
    "infra/terra/examples",
    "libreoffice"
    ],
    "https://gitlab.org/claims/groups/owner": [
    "ocs-infra",
    "go-shared",
    "appstore",
    "auth",
    "cdn",
    "cloud",
    "js-shared",
    "emm",
    "pkgrepo",
    "sfc-sandbox",
    "push",
    "mt",
    "omp-qa",
    "ocs-qa",
    "mer-core",
    "nemomobile",
    "mer-hybris",
    "sailfishos",
    "customer-omp",
    "omp_alldevops",
    "mer-packages",
    "mer-crosshelpers",
    "mer-tools",
    "openmobileplatform",
    "show",
    "test_group_x",
    "omp-sdk",
    "qt",
    "omp-browser",
    "l10n",
    "popul-census",
    "devops",
    "native-components",
    "libhybris",
    "non-oss",
    "release_management",
    "external_portal",
    "external_devdoc"
    ],
    "name": "Vladimir Skesov",
    "nickname": "v.skesov",
    "picture": "https://git.omprussia.ru/uploads/-/system/user/avatar/345/avatar.png",
    "profile": "https://git.omprussia.ru/v.skesov",
    "sub": "2222",
    "sub_legacy": "1231asd321as321a23d1a231"
    }
```

And managed group with fake filter

![Screenshot from 2023-01-18 21-13-26](https://user-images.githubusercontent.com/12987308/213261800-4f45a556-c474-42db-83c9-2acb97106962.png)

But all registered users in members
![Screenshot from 2023-01-18 21-13-34](https://user-images.githubusercontent.com/12987308/213262000-af985a96-6a85-4b5f-b20e-873f342607b5.png)

```bash
boundary managed-groups read -id mgoidc_qjkJ6tXdqY

Managed Group information:
  Auth Method ID:      amoidc_nQ3L3GlbX9
  Created Time:        Wed, 18 Jan 2023 21:01:23 MSK
  ID:                  mgoidc_qjkJ6tXdqY
  Name:                Test
  Type:                oidc
  Updated Time:        Wed, 18 Jan 2023 21:13:14 MSK
  Version:             5

  Scope:
    ID:                global
    Name:              global
    Type:              global

  Authorized Actions:
    no-op
    read
    update
    delete

  Attributes:
    Filter:            email == "not_in_account"
```

**Comment by jefferai** [comment]
Seems like the same issue -- the managed group should not match anything and does not display anything on the CLI but is in the UI.

**Comment by xingluw** [comment]
@RobertSkawinski Could you send me an email at xinglu@hashicorp.com to set up some time to call next week?

**Comment by xingluw** [comment]
Fixed in Boundary 0.12
